editor/editor_properties.cpp

```diff
@@ -462,10 +462,26 @@ void EditorPropertyPath::_set_read_only(bool p_read_only) {
 }
 
 void EditorPropertyPath::_path_selected(const String &p_path) {
-	emit_changed(get_edited_property(), p_path);
+	String full_path = p_path;
+	ResourceUID::ID id = ResourceLoader::get_resource_uid(full_path);
+
+	if (id != ResourceUID::INVALID_ID) {
+		full_path = ResourceUID::get_singleton()->id_to_text(id);
+	}
+
+	emit_changed(get_edited_property(), full_path);
 	update_property();
 }
 
+String EditorPropertyPath::_get_path_text() {
+	String full_path = get_edited_property_value();
+	if (full_path.begins_with("uid://")) {
+		full_path = ResourceUID::get_singleton()->get_id_path(ResourceUID::get_singleton()->text_to_id(full_path));
```

Comment:
This can be simplified to
```suggestion
		return ResourceUID::get_singleton()->get_id_path(ResourceUID::get_singleton()->text_to_id(full_path));
```
and you can make `full_path` const.

Reply:
I was about to amend those changes in but I'm not convinced this improves the code quality much. I quite like the pattern where we have a single `return` statement at the end and just do conditional changes to the return value.